app/models/post.rb

```diff
@@ -1,15 +1,17 @@
 class Post < ApplicationRecord
-  belongs_to :user, foreign_key: :author_id
-  has_many :comments
-  has_many :likes
-  after_save :update_posts_counter
-  # A method that updates the posts counter for a user.
-  def update_posts_counter
-    user.increment!(:posts_counter)
+  belongs_to :user, class_name: :User, foreign_key: 'author_id'
+  has_many :likes, dependent: :destroy
+  has_many :comments, dependent: :destroy
+  after_save :increment_counter
+  validates :title, presence: true
+  validates :text, presence: true, length: { maximum: 250 }
+  validates :comments_counter, :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
```

Comment:
Please note comments_counter and likes_counter must be an integer greater than zero only.

- [ ] Kindly update this validation

Reply:
Done✔